app/experimenter/experiments/tasks.py

```diff
@@ -123,8 +125,61 @@ def update_experiment_bug_task(user_id, experiment_id):
 @app.task
 @metrics.timer_decorator("update_experiment_info.timing")
 def update_experiment_info():
-    metrics.incr("update_experiment_info.started")
-    logger.info("Updating experiment info")
+    update_ready_to_ship_experiments()
+    update_launched_experiments()
+
+
+@app.task
+@metrics.timer_decorator("update_ready_to_ship_experiments.timing")
+def update_ready_to_ship_experiments():
+    metrics.incr("update_ready_to_ship_experiments.started")
+    logger.info("Updating ready to ship experiment")
+
+    ready_to_ship_experiments = Experiment.objects.filter(status=Experiment.STATUS_SHIP)
+    for experiment in ready_to_ship_experiments:
+        try:
+            logger.info("Updating Experiment: {}".format(experiment))
+            recipe_data = normandy.get_recipe_list(
+                params={"experimenter_slug": experiment.slug}
+            )
+
+            if len(recipe_data):
+                recipe_ids = [r["id"] for r in recipe_data]
+                # sort to get oldest id to be primary
+                recipe_ids.sort()
+                changed_data = {
+                    "normandy_id": recipe_ids[0],
+                    "other_normandy_ids": recipe_ids[1:],
+                    "status": Experiment.STATUS_ACCEPTED,
+                }
+
+                old_serialized_exp = ChangeLogSerializer(experiment).data
+                Experiment.objects.filter(id=experiment.id).update(**changed_data)
+                experiment = Experiment.objects.get(slug=experiment.slug)
+                new_serialized_exp = ChangeLogSerializer(experiment).data
+                normandy_user = settings.NORMANDY_DEFAULT_CHANGELOG_USER
+                default_user, _ = get_user_model().objects.get_or_create(
+                    email=normandy_user, username=normandy_user
+                )
+
+                generate_change_log(
+                    old_serialized_exp,
+                    new_serialized_exp,
+                    experiment,
+                    changed_data,
+                    default_user,
+                )
```

Comment:
Can we move all this into a model method?  I feel like we have to do this pattern a whole bunch and it's a bit verbose to have to do all these steps here explicitly.

Reply:
i moved this into changelog utils instead, since moving it to models.py creates a circular import problem